akka-actor/src/main/scala/akka/serialization/AsyncSerializer.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.serialization
+
+import akka.actor.ExtendedActorSystem
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, Future }
+
+/**
+ * Serializer that supports async serialization.
+ *
+ * Only used for Akka persistence journals that explicitly support async serializers.
+ *
+ * Implementations should typically extend [[AsyncSerializerWithStringManifest]] that
+ * delegates synchronous calls to their async equivalents.
+ */
+trait AsyncSerializer {
+  def toBinaryAsync(o: AnyRef): Future[Array[Byte]]
+
+  def fromBinaryAsync(bytes: Array[Byte], manifest: String): Future[AnyRef]
```

Comment:
hope this is considered an advanced (library author) feature so that we don't have to add same with `CompletionStage` signature

Reply:
Hopefully we're the only author :)